### PR TITLE
Get metric "polls/request" for benchmarks

### PR DESCRIPTION
**Pull request by lyuxuan**

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
Performance differences noted:
Benchmark     atm_add_per_iteration
------------  -----------------------
BM_EmptyCore  +16%
```

File: src/core/lib/surface/completion_queue.c [hunk 1]
```diff
@@ -592,6 +598,7 @@ grpc_event grpc_completion_queue_next(grpc_completion_queue *cc,
       gpr_mu_lock(cc->mu);
       continue;
     } else {
+      gpr_atm_no_barrier_fetch_add(&cc->num_polls, 1);
```
Comment (ctiller) [inline src/core/lib/surface/completion_queue.c]: Suggest making this a normal int (no need for an atomic here) and just locking when we do the get

File: test/cpp/qps/client_async.cc [hunk 2]
```diff
@@ -54,6 +54,10 @@
 #include "test/cpp/qps/usage_timer.h"
 #include "test/cpp/util/create_test_channel.h"
 
+extern "C" {
```
Comment (vjpai) [inline test/cpp/qps/client_async.cc]: Can you move this extern "C" protection into the included header file? It generally makes more sense there (particularly if the underlying implementation ends up changing to C++)

File: src/core/lib/surface/completion_queue.c (hunk 1 above)
Comment (vjpai) [inline src/core/lib/surface/completion_queue.c]: Agreed. We're already holding the lock at this point, and we are going to call this increment way more than we ever call the "get" function. So better just to make this an int and then put a lock there.

**Comment by grpc-testing** [comment]
```
Performance differences noted:
Benchmark     atm_add_per_iteration
------------  -----------------------
BM_EmptyCore  +16%
```

**Comment by grpc-testing** [comment]
```
No significant performance differences
```

**Comment by lyuxuan** [comment]
ping @vjpai

**Comment by lyuxuan** [comment]
ping @vjpai

File: test/cpp/qps/client.h [hunk 3]
```diff
@@ -54,6 +54,10 @@
 #include "test/cpp/qps/usage_timer.h"
 #include "test/cpp/util/create_test_channel.h"
 
+extern "C" {
+#include "src/core/lib/surface/completion_queue.h"
```
Comment (vjpai) [inline test/cpp/qps/client.h]: Oh, I think my last comment was not clear. I meant that this extern "C" business should go inside `completion_queue.h` itself (within an `#ifdef __cplusplus` like the way it is used in many other files.)

File: test/cpp/qps/server.h [hunk 4]
```diff
@@ -44,12 +44,17 @@
 #include "test/core/util/port.h"
 #include "test/cpp/qps/usage_timer.h"
 
+extern "C" {
+#include "src/core/lib/surface/completion_queue.h"
+}
```
Comment (vjpai) [inline test/cpp/qps/server.h]: ditto the above

**Comment by grpc-testing** [comment]
```
No significant performance differences
```

**Comment by lyuxuan** [comment]
@vjpai fixed the extern c location.

**Comment by lyuxuan** [comment]
known failures: #9088, #11105, #11116

**Pull request merged**